src/LocalField/ResidueRing.jl

```diff
@@ -0,0 +1,357 @@
+################################################################################
+#
+#  Parent type etc
+#
+################################################################################
+
+parent_type(::Type{LocalFieldValuationRingResidueRingElem{S, T}}) where {S, T} = LocalFieldValuationRingResidueRing{S, T}
+elem_type(::Type{LocalFieldValuationRingResidueRing{S, T}}) where {S, T} = LocalFieldValuationRingResidueRingElem{S, T}
+is_domain_type(::Type{<: LocalFieldValuationRingResidueRing}) = false
+is_exact_type(::Type{<: LocalFieldValuationRingResidueRing}) = true
+
+################################################################################
+#
+#  Field access
+#
+################################################################################
+
+_valuation_ring(R::LocalFieldValuationRingResidueRing) = R.R
+_field(R::LocalFieldValuationRingResidueRing) = _valuation_ring(R).Q
+_exponent(R::LocalFieldValuationRingResidueRing) = R.k
+
+base_ring(R::LocalFieldValuationRingResidueRing) = Union{}
+base_ring_type(::Type{<: LocalFieldValuationRingResidueRing}) = typeof(Union{})
+
+parent(a::LocalFieldValuationRingResidueRingElem) = a.parent
+data(a::LocalFieldValuationRingResidueRingElem) = a.a
+lift(a::LocalFieldValuationRingResidueRingElem) = _valuation_ring(parent(a))(data(a))
+
+################################################################################
+#
+#  Basic functionality
+#
+################################################################################
+
+characteristic(R::LocalFieldValuationRingResidueRing) = prime(_field(R))^_exponent(R)
+
+zero(R::LocalFieldValuationRingResidueRing) = R()
```

Comment:
unless the zerois exact, it needs a precision....

Reply:
This is `Zp/p^k`, so the zero is the exact zero of this ring.

Reply:
I mean, this one is exact because it is the zero of the residue ring?